### ISSUE TITLE
Conv2d custom weights

### ISSUE THREAD

**Issue by utkusaglm** [keras-team/keras#18893]
```
x = np.random.rand(4, 10, 10, 128)
y = keras.layers.Conv2D(32, 3, activation='relu',kernel_initializer='glorot_uniform')
print(y.weights)
y.add_weight(shape =(3, 3, 128, 32),initializer='glorot_uniform', name='kernelA',trainable=True)
print(y.weights)
z = y(x)
print(y.weights)
```
Output:
```
"[]
[<KerasVariable shape=(3, 3, 128, 32), dtype=float32, path=conv2d_1/kernelA>]
[<KerasVariable shape=(3, 3, 128, 32), dtype=float32, path=conv2d_1/kernelA>, <KerasVariable shape=(3, 3, 128, 32), dtype=float32, path=conv2d_1/kernel>, <KerasVariable shape=(32,), dtype=float32, path=conv2d_1/bias>]"
```

So I wanted to use my kernels/weights, but I cannot since Keras conv2d creates another kernel by default after It runs with input. This becomes problematic in train cases since it just sees the kernel it created. How could I overcome this issue or this feature?

Also, when we want to implement it as a customer layer: `customconv2d(keras.layers.Layer)` with init, build, and call? How should we add our custom weights?

I couldn't see anything about it on the documentation; it was just about the initializer. Also, for custom layers, there was just an example for linear.

**Comment by sachinprasadhs** [comment]
Hi, 

I was able to reproduce the issue using Keras 3, please find the attached Gist [here](https://colab.sandbox.google.com/gist/sachinprasadhs/e507a69f288fdb2ad780c6bcd28f1e93/18893.ipynb) for reference. Thanks!

**Comment by MoritzMo123** [comment]
This code looks to me like it is working as expected.

Linie by line my understanding is as follows:
1. dummy data
2. You create a vanilla Conv2D layer (not build yet)
3. First print: [] -> nothing has been built yet, so no weights are present
4. You manually add a weight called KernelA to the layer.
5. Second print: [KernelA] -> the weight you added appears in weights, however the layer itself still has not been and cannot be build as not all input dimensions are known.
6. You build the layer by passing an input through it. Now all dimensions are known and the kernel tensor can be initialized.
7. Third print: [KernelA, Kernel, Bias] -> the build function of the Conv2D layer added a kernel and bias to the weights. These Kernel and Bias weights are used in the Conv2D layer calculations.
 
Note that the .weights property just shows all the weights that have been added, it does not indicate if the weights in the list are used or not (however, the optimizer would warn you that there is no gradient for your kernelA, if you were to train the layer). Custom behavior with custom weights requires modifications to the call statement, which would require a custom layer. The custom layer can be created using standard inheritance and is well documented in the [api docs for base layer](https://keras.io/api/layers/base_layer/#weights-property) that you already referenced. While the doc page references a linear layer in particular, the example should provide all the information you need to create any custom layer with custom weights and custom functionality.

**Comment by utkusaglm** [comment]
Hey @MoritzMo123, thanks a lot for your detailed answer and for explaining it line by line. I wanted to ask about this feature to clarify how it works. Since, as I thought, it is not an issue, I'm closing it.
